pkg/asset/agent/image/agentpxefiles.go

```diff
@@ -120,9 +123,22 @@ func (a *AgentPXEFiles) PersistToFile(directory string) error {
 		return err
 	}
 	defer kernelReader.Close()
-	err = a.copy(agentVmlinuzFile, kernelReader)
-	if err != nil {
-		return err
+
+	if a.cpuArch == arch.RpmArch(types.ArchitectureARM64) {
```

Comment:
It could be useful to have a switch capturing the known archs, and failing for all the others